### PR TITLE
Keep as much span info as possible

### LINKED ISSUE #247: Most of the generated code should carry span info about its origins
```
error[E0277]: the trait bound `u32: std::convert::From<&str>` is not satisfied

  --> tests/skip.rs:98:14

   |
98 |     #[derive(StructOpt, Debug, PartialEq)]
   |              ^^^^^^^^^ the trait `std::convert::From<&str>` is not implemented for `u32`

   = help: the following implementations were found:

             <u32 as std::convert::From<bool>>
             <u32 as std::convert::From<char>>
             <u32 as std::convert::From<std::net::Ipv4Addr>>
             <u32 as std::convert::From<std::num::NonZeroU32>>

           and 2 others

   = note: required by `std::convert::From::from`
```

Errors like this should point to the field

### PR DESCRIPTION
**Pull request by CreepySkeleton**
Closes #247 
Work-in-progress, just so you could state your opinion on this changes

Em, fix to #156 is also here, I can extract it to a separate PR if you want

### CODE REVIEW

**Comment by CreepySkeleton** [comment]
@TeXitoi Pretty much ready, waiting on review

**Comment by CreepySkeleton** [comment]
ping @TeXitoi , sorry for being annoying but I need your opinion to move on

File: structopt-derive/src/attrs.rs [hunk 1]
```diff
@@ -77,28 +81,91 @@ pub struct Attrs {
...
     casing: Sp<CasingStyle>,
     methods: Vec<Method>,
-    parser: Sp<(Sp<Parser>, TokenStream)>,
-    author: Option<(Ident, LitStr)>,
-    about: Option<(Ident, LitStr)>,
-    version: Option<(Ident, LitStr)>,
+    parser: Sp<Parser>,
+    author: Option<(Method)>,
+    about: Option<(Method)>,
+    version: Option<(Method)>,
```
Comment (TeXitoi) [inline structopt-derive/src/attrs.rs]: Useless `()`

**Comment by TeXitoi** [comment]
for #156 a separate commit would be great.

File: structopt-derive/src/attrs.rs (hunk 1 above)
Comment (CreepySkeleton) [reply structopt-derive/src/attrs.rs]: Erased

**Comment by CreepySkeleton** [comment]
Well, that's final line. I'll go work on AST implementation, should be done in a week or so (I'm kind of overloaded by work/life stuff right now, sorry)

**Comment by CreepySkeleton** [comment]
ping @TeXitoi

**Comment by CreepySkeleton** [comment]
cc @TeXitoi

**Comment by CreepySkeleton** [comment]
> Changelog entry and I merge.

@TeXitoi 
I don't think there's much to put in changelog about #247. These changes are purely cosmetical

**Comment by TeXitoi** [comment]
There is an error message change that fail the nightly build :-(

**Comment by CreepySkeleton** [comment]
@TeXitoi They look **identical**. I'll investigate in few hours

**Comment by TeXitoi** [comment]
There is an extra message after the ^^^^^^^^

**Comment by CreepySkeleton** [comment]
Fucking nightly! Will fix

**Comment by TeXitoi** [comment]
Thanks.

**Pull request merged**